internal/provider/provider_test.go

```
@@ -1,41 +1,36 @@
 }
 
 func testAccPreCheck(t *testing.T) {
 	if v := os.Getenv("ABLY_ACCOUNT_TOKEN"); v == "" {
 		t.Fatal("ABLY_ACCOUNT_TOKEN must be set for acceptance tests")
 	}
 }
+
+func TestProvider(t *testing.T) {
+	// Just test that the provider type exists
+	p := &AblyProvider{}
+	if p == nil {
```

golangci-lint (1.64.8)

SA4031: this nil check is never true

(staticcheck)